### PR TITLE
Synapse Workspace Activation with CMK

### PR DESCRIPTION
**Pull request by MoAhuja**
Fixes: #12348

- Added new synapse_workspace_key resource type to support activation of a workspace encryption key
- Updated synapse_workspace to support naming of the CMK. Currently implementation defaults to "cmk" which makes it challenging to rotate encryption keys.

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks for the pr @MoAhuja - looks like we have some test failures to fix:
```
------- Stdout: -------
=== RUN   TestAccSynapseWorkspace_azdo
=== PAUSE TestAccSynapseWorkspace_azdo
=== CONT  TestAccSynapseWorkspace_azdo
    testcase.go:88: Step 1/1 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # azurerm_synapse_workspace.test will be updated in-place
          ~ resource "azurerm_synapse_workspace" "test" {
              + customer_managed_key_name            = "cmk"
                id                                   = "/subscriptions/*******/resourceGroups/acctestRG-synapse-210831222712997048/providers/Microsoft.Synapse/workspaces/acctestsw210831222712997048"
                name                                 = "acctestsw210831222712997048"
                # (11 unchanged attributes hidden)
        
                # (1 unchanged block hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccSynapseWorkspace_azdo (741.82s)
FAIL
```

File: internal/services/synapse/synapse_workspace_key_resource.go [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+				Required: true,
+			},
+
+			"customer_managed_key_versionless_id": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: keyVaultValidate.VersionlessNestedItemId,
+			},
+
+			"is_active_cmk": {
```
[the remark this answers is not in the document]
Comment (MoAhuja) [reply internal/services/synapse/synapse_workspace_key_resource.go]: Adjusted

**Comment by MoAhuja** [comment]
@katbyte  Thanks for the review. I've modified the synapse_workspace resource's property structure to bundle up key_name and key_versionless_id into a new type. This seemed like the only way to support a model where key_name could be optional and defaulted to "cmk" while making sure it's only ever used if key_versionless_id is set.

Called this out here because this is breaking change from the previous version of the synapse workspace  resource type and I'm not sure how those typically get handled as part of provider releases.

**Review by katbyte — APPROVED** [review]
Thanks @MoAhuja - LGTM 🏗️

**Pull request merged**